### PR TITLE
Upgrade to BuckleScript 3.0

### PR DESCRIPTION
**Pull request by snird**
Upgrade to the new BS release.
This includes dropping usage of deprecated `Js.boolean` everywhere.

It is a breaking change, since there were `boolean` and `bool` methods on decoder, and `boolean` is removed now.

Tests are passing, and I personally used it on my codebase successfully.

I divided it into easy to follow commits, first initial upgrade to BS 3.0 and build with it (as it builds differently) and then upgrading decoder and then encoder. So it is easy to follow changes within each commit.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
   "description": "Compositional JSON encode/decode library for BuckleScript",
   "main": "index.js",
   "scripts": {
-    "build": "bsb -make-world",
-    "start": "bsb -make-world -w",
+    "build": "./node_modules/.bin/bsb -make-world",
```
Comment (snird) [inline package.json]: did that so it'll build with the local bucklescript for this package, and not be overridden by global installation

Comment (glennsl) [reply package.json]: I actually think I prefer to have it as it is, as I then only have to upgrade the global install and will get a warning if there's a version mismatch between local and global. Using this, if the local install is out of date, the result will just be wrong.

File: src/Json_decode.mli [hunk 2]
```diff
@@ -15,26 +15,6 @@ type 'a decoder = Js.Json.t -> 'a
 
 exception DecodeError of string
 
-val boolean : Js.boolean decoder
```
Comment (glennsl) [inline src/Json_decode.mli]: This should be deprecated first, before being removed in a later major release.

**Comment by snird** [comment]
Got it. I wanted an upgrade anyway for my internal use, so figured I should make a proper PR.
Anyway, hope this is more help than struggle (:

**Comment by glennsl** [comment]
Closing, as I'm pretty sure this is obsolete now. Thanks for submitting though :)

**Pull request closed without merging**